pkg/machinery/client/client.go

```diff
@@ -171,6 +173,14 @@ func (c *Client) GetConn(ctx context.Context, opts ...grpc.DialOption) (*grpc.Cl
 
 	var target string
 
+	for i, endpoint := range endpoints {
+		if net.AddressContainsPort(endpoint) {
+			endpoints[i] = net.FormatAddress(endpoint)
```

Comment:
please let's try to avoid using this package, as it's confusing, and current behavior is wrong 

we should instead do the following: https://pkg.go.dev/net#SplitHostPort